src/main/java/org/dcsa/ebl/service/impl/TransportDocumentServiceImpl.java

```diff
@@ -1,40 +1,182 @@
+    TransportDocumentSummary transportDocumentSummary =
+        transportDocumentMapper.transportDocumentToTransportDocumentSummary(transportDocument);
+
+    return shippingInstructionRepository
+        .findById(transportDocumentSummary.getShippingInstructionID())
+        .switchIfEmpty(
+            Mono.error(
+                ConcreteRequestErrorMessageException.internalServerError(
+                    "No shipping instruction was found with ID: "
+                        + transportDocument.getShippingInstructionID())))
+        .flatMap(
+            shippingInstruction -> {
+              transportDocumentSummary.setDocumentStatus(shippingInstruction.getDocumentStatus());
+              return shippingInstructionRepository
+                  .findCarrierBookingReferenceByShippingInstructionID(
+                      shippingInstruction.getShippingInstructionID())
+                  .collectList()
+                  .doOnNext(transportDocumentSummary::setCarrierBookingReferences)
+                  .thenReturn(transportDocumentSummary);
+            })
+        .flatMap(
+            ignored -> {
+              if (transportDocument.getIssuer() == null) return Mono.just(transportDocumentSummary);
+              return carrierRepository
+                  .findById(transportDocument.getIssuer())
+                  .switchIfEmpty(
+                      Mono.error(
+                          ConcreteRequestErrorMessageException.internalServerError(
+                              "No carrier found with issuer ID: " + transportDocument.getIssuer())))
+                  .flatMap(
+                      carrier -> {
+                        if (carrier.getSmdgCode() != null) {
+                          transportDocumentSummary.setIssuerCodeListProvider(
+                              CarrierCodeListProvider.SMDG);
+                          transportDocumentSummary.setIssuerCode(carrier.getSmdgCode());
+                        } else if (carrier.getNmftaCode() != null) {
+                          transportDocumentSummary.setIssuerCodeListProvider(
+                              CarrierCodeListProvider.NMFTA);
+                          transportDocumentSummary.setIssuerCode(carrier.getNmftaCode());
+                        } else {
+                          return Mono.error(
+                              ConcreteRequestErrorMessageException.invalidParameter(
+                                  "Unsupported carrier code list provider."));
+                        }
+                        return Mono.just(transportDocumentSummary);
+                      });
+            });
+  }
+
+  private Mono<Booking> getBooking(String carrierBookingReference, String shippingInstructionID) {
+    // Don't use ServiceClass - use Repository directly in order to throw internal error if
+    // BookingReference does not exist.
+    return bookingRepository
+        .findByCarrierBookingRequestReference(carrierBookingReference)
```

Comment:
Why can't org.dcsa.core.events.edocumentation.service.ShipmentService#findByShippingInstructionID be used in addition with a switchIfEmpty()

Reply:
method was unused, so I've removed it.